src/include/migraphx/op/nonmaxsuppression.hpp

```diff
@@ -219,30 +226,42 @@ struct nonmaxsuppression
 
     // filter boxes below score_threshold
     template <class T>
-    std::priority_queue<std::pair<double, int64_t>>
+    std::vector<std::pair<double, int64_t>>
     filter_boxes_by_score(T scores_start, std::size_t num_boxes, double score_threshold) const
     {
-        std::priority_queue<std::pair<double, int64_t>> boxes_heap;
+        std::vector<std::pair<double, int64_t>> boxes_heap;
         auto insert_to_boxes_heap =
-            make_function_output_iterator([&](const auto& x) { boxes_heap.push(x); });
+            make_function_output_iterator([&](const auto& x) { boxes_heap.push_back(x); });
         int64_t box_idx = 0;
-        transform_if(
-            scores_start,
-            scores_start + num_boxes,
-            insert_to_boxes_heap,
-            [&](auto sc) {
-                box_idx++;
-                return sc >= score_threshold;
-            },
-            [&](auto sc) { return std::make_pair(sc, box_idx - 1); });
+
+        if(score_threshold > 0.0)
+        {
+            transform_if(
+                scores_start,
+                scores_start + num_boxes,
+                insert_to_boxes_heap,
+                [&](auto sc) {
+                    box_idx++;
+                    return sc >= score_threshold;
+                },
+                [&](auto sc) { return std::make_pair(sc, box_idx - 1); });
+        }
+        else
+        { // score is irrelevant, just push into boxes_heap and make a score-index pair
```

Comment:
not sure if score_threshold negative is possible or not, but even in that case we want to only select boxes that have scores higher than the (negative) threshold.

Reply:
Score can't be negative and is bounded between 0-1 since its statistical confidence the box is in a certain class. In this case it pre filters the boxes based on the confidence before we perform the IOU filtering.

Reply:
There can be a case here to probably leverage execution::par for the transform for the unfiltered (score = 0) case and probably copy_if for the score > 0 case to do the copy in parallel...

Reply:
Please get rid of this if/else clause. Let it all be handled in a uniform way, particularly when the score is between 0 and 1 (inclusive).